src/modules/src/Eryph.Modules.Controller/Compute/UpdateCatletSaga.cs

```diff
@@ -2,287 +2,373 @@
-                        {
-                            VMId = r.Inventory.VMId,
-                            CatletId = Data.CatletId,
-                            CatletName = data.Name,
-                            MachineMetadata = r.MachineMetadata,
-                        }).AsTask();
-                    },
-                    None: () => Fail(new ErrorData
-                    { ErrorMessage = $"Could not find virtual catlet with catlet id {Data.CatletId}" })
-                );
-
+                Projects = [Data.Data.ProjectId]
             });
-        }
+        });
+    }
 
+    public Task Handle(OperationTaskStatusEvent<UpdateNetworksCommand> message)
+    {
+        return FailOrRun(message, () => Complete());
+    }
 
-        public Task Handle(OperationTaskStatusEvent<UpdateCatletConfigDriveCommand> message)
+    protected override void CorrelateMessages(ICorrelationConfig<EryphSagaData<UpdateCatletSagaData>> config)
+    {
+        base.CorrelateMessages(config);
+
+        config.Correlate<OperationTaskStatusEvent<ValidateCatletConfigCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+        config.Correlate<OperationTaskStatusEvent<ResolveCatletConfigCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+        config.Correlate<OperationTaskStatusEvent<PrepareGeneCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+        config.Correlate<OperationTaskStatusEvent<UpdateCatletNetworksCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+        config.Correlate<OperationTaskStatusEvent<UpdateCatletVMCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+        config.Correlate<OperationTaskStatusEvent<UpdateCatletConfigDriveCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+        config.Correlate<OperationTaskStatusEvent<UpdateNetworksCommand>>(
+            m => m.InitiatingTaskId, d => d.SagaTaskId);
+    }
+
+    internal static Validation<Error, Seq<GeneIdentifierWithType>> FindRequiredGenes(
+        CatletConfig catletConfig) =>
+        CatletGeneCollecting.CollectGenes(catletConfig)
+            .Map(l => l.Filter(c => c.GeneIdentifier.GeneName != GeneName.New("catlet")));
+
+    internal static Either<Error, (CatletConfig Config, CatletConfig BredConfig)> PrepareConfigs(
+    CatletConfig config,
+    CatletMetadata metadata,
+    HashMap<GeneSetIdentifier, GeneSetIdentifier> resolvedGeneSets,
+    HashMap<GeneSetIdentifier, CatletConfig> parentConfigs) =>
+    from resolvedConfig in CatletGeneResolving.ResolveGeneSetIdentifiers(
+            config, resolvedGeneSets)
+        .MapLeft(e => Error.New("Could not resolve genes in catlet config.", e))
+    from breedingResult in CatletPedigree.Breed(
+            config, resolvedGeneSets, parentConfigs)
+        .MapLeft(e => Error.New("Could not breed the catlet.", e))
+    // After the catlet was created, the fodder and variables can no longer be changed.
```

Comment:
The user experience for update is a bit inconsistant in this case. Update-Catlet asks for variables, even if not supported to change them. So we should remove variable input for update catlet (the only use case I see here is that catlet was not started in between, but this is something we could ignore as edge.

Reply:
I have removed the variable input from the `Update-Catlet` Cmdlet.